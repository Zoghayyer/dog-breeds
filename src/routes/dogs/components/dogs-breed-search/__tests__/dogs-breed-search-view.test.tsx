import React from 'react';
import { shallow } from 'enzyme';
import DogsBreedSearchView, {
  Props,
} from '../dogs-breed-search-view';

describe('(View) DogsBreedSearchView', () => {
  let component;
  const defaultProps: Props = {
    dogsBreedValue: '',
    handleChange: () => ({}),
    handleSubmit: () => ({}),
    hasError: false,
  };

  const shallowDogsBreedSearchView = (extendedProps: Partial<Props> = {}) => {
    component = shallow(<DogsBreedSearchView {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowDogsBreedSearchView());

  it('renders the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the invalid search message', () => {
    shallowDogsBreedSearchView({ hasError: true });
    expect(component).toMatchSnapshot();
  });
});
