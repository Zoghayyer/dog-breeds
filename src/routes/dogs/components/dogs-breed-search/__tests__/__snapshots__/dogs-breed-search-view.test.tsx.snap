// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(View) DogsBreedSearchView renders the component 1`] = `
<div
  className="mt-4"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-md-10 search-input-section"
      >
        <input
          aria-describedby="dogs breed input search"
          className="search-input"
          onChange={[Function]}
          placeholder="Search for your favorite breed of dogs..."
          type="text"
          value=""
        />
      </div>
      <div
        className="col-md-2 search-button-section"
      >
        <Button
          className="search-button"
          size="sm"
          theme="primary"
          type="submit"
        >
          <Icon
            alt="search icon"
            className="mr-3 search-icon"
            icon="searchIcon"
          />
          <span
            className="search-message"
          >
            Search
          </span>
        </Button>
      </div>
    </div>
  </form>
</div>
`;

exports[`(View) DogsBreedSearchView renders the invalid search message 1`] = `
<div
  className="mt-4"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-md-10 search-input-section"
      >
        <input
          aria-describedby="dogs breed input search"
          className="search-input"
          onChange={[Function]}
          placeholder="Search for your favorite breed of dogs..."
          type="text"
          value=""
        />
        <div
          className="invalid-search"
        >
          Sorry! We couldn't find the dog breed that you're looking for.
        </div>
      </div>
      <div
        className="col-md-2 search-button-section"
      >
        <Button
          className="search-button"
          size="sm"
          theme="primary"
          type="submit"
        >
          <Icon
            alt="search icon"
            className="mr-3 search-icon"
            icon="searchIcon"
          />
          <span
            className="search-message"
          >
            Search
          </span>
        </Button>
      </div>
    </div>
  </form>
</div>
`;
